src/components/Checkbox/Group.js

```diff
@@ -1,18 +1,26 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Label } from '../Label';
 import styles from './Checkbox.module.scss';
 
-const Group = ({ children }) => (
+const Group = ({ children, label, required }) => (
   <div className={styles.Group}>
+    {label && <Label className={styles.GroupLabel}>{label}{required && '*'}</Label>}
     { children }
   </div>
 );
 
+Group.defaultProps = {
+  required: false
+};
+
 Group.propTypes = {
   children: PropTypes.oneOfType([
     PropTypes.arrayOf(PropTypes.node),
     PropTypes.node
-  ]).isRequired
+  ]).isRequired,
+  label: PropTypes.string,
```

Comment:
May want label to also work with react components, for example a string with an UnstyledLink